KalturaWowzaServer/src/main/java/com/kaltura/media_server/services/KalturaEntryDataPersistence.java

```diff
@@ -23,49 +23,58 @@ public static void setAppInstance(IApplicationInstance appInstance) {
 	}
 
 	public static void entriesMapCleanUp() {
-	    Timer cleanUpTimer = new Timer();
-	    cleanUpTimer.schedule(new TimerTask() {
-            @Override
-            public void run() {
-				CleanUp();
-            }
-        }, Constants.KALTURA_ENTRY_PERSISTENCE_CLEANUP_START);
-        logger.info("Persistence hash map cleanup will start in " + Constants.KALTURA_ENTRY_PERSISTENCE_CLEANUP_START / 1000 + " seconds");
+        // Do not perform entries cleanup more than once a minute, to prevent a load of timer tasks running at the same time.
+	    if (System.currentTimeMillis() - lastMapCleanUp > Constants.KALTURA_TIME_BETWEEN_PERSISTENCE_CLEANUP) {
+            Timer cleanUpTimer = new Timer();
+            cleanUpTimer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    CleanUp();
+                }
+            }, Constants.KALTURA_ENTRY_PERSISTENCE_CLEANUP_START);
+            logger.debug("Persistence hash map cleanup will start in " + Constants.KALTURA_ENTRY_PERSISTENCE_CLEANUP_START / 1000 + " seconds");
+        }
+        else {
+            logger.debug("Persistence cleanup was called less than " + Constants.KALTURA_TIME_BETWEEN_PERSISTENCE_CLEANUP / 1000 + " seconds ago. Ignoring call!");
+        }
 	}
 
 	private static void CleanUp() {
-        synchronized (entryIdToKalturaLiveEntryMap) {
-            // Do not perform entries cleanup more than once a minute, to prevent a load of timer tasks running at the same time.
-            long currentTime = System.currentTimeMillis();
-            if (currentTime - lastMapCleanUp > Constants.KALTURA_TIME_BETWEEN_PERSISTENCE_CLEANUP) {
-                try {
-                    Set<String> playingEntriesList = Utils.getEntriesFromApplication(_appInstance);
-                    Set<String> hashedEntriesList = entryIdToKalturaLiveEntryMap.keySet();
-                    long currTime = System.currentTimeMillis();
-                    for (String entry : hashedEntriesList) {
-                        // Check start time to avoid a race between the thread adding the entry to the map and this
-                        // current thread that wants to remove it. Worst case scenario entry will ve erased in the next run.
-                        long startTime = (long) getPropertyByEntry(entry, Constants.KALTURA_ENTRY_START_TIME);
-                        if ((!playingEntriesList.contains(entry)) && (currTime - startTime > Constants.KALTURA_PERSISTENCE_DATA_MIN_ENTRY_TIME)) {
-                            logger.info("Entry " + entry + " no longer exists. Removing it from Persistence Hash Map");
+	    try {
+            synchronized (entryIdToKalturaLiveEntryMap) {
```

Comment:
where do you update lastMapCleanup? need to update once in entriesMapCleanUp and once in CleanUp with sync on lastMapCleanup.